client/src/leap/soledad/client/api.py

```diff
@@ -70,6 +71,26 @@
 """
 SOLEDAD_CERT = None
 
+# A whitelist of modules from where to collect plugins dynamically.
+# For the moment restricted to leap namespace, but the idea is that we can pass
+# other "trusted" modules as options to the initialization of soledad.
+PLUGGABLE_LEAP_MODULES = ('mail', 'keymanager')
+
+
+def collect_plugins(interface):
+    """
+    Traverse a whitelist of modules and collect all the plugins that implement
+    the passed interface.
+    """
+    plugins = []
+    for namespace in PLUGGABLE_LEAP_MODULES:
+        try:
+            module = __import__('leap.%s.plugins' % namespace, fromlist='.')
+            plugins = plugins + list(getPlugins(interface, module))
```

Comment:
each module will be able to drop its plugins on its plugin/ folder, for any pluggable interface that other leap modules implement, this soledad interface being the first.

each leap module could also provide (and we *should* document it properly) interfaces for hooks that other modules can write plugins for. I'm pointing now to add a very similar hook to  leap.mail to allow plugins to "register" to a post-mail-decryption hook